### PR TITLE
Increase BuildPodControllerTest timeout

### PR DESCRIPTION
**Pull request by csrwng**
Increases the build controller test timeout to give the informer a chance to send pod update events.

fixes https://github.com/openshift/origin/issues/13650 (we hope)

### CODE REVIEW

**Comment by csrwng** [comment]
@bparees so I looked at the logs a little bit longer and I came to the conclusion that I won't gain much from adding another event listener for the pod informer. The pod disruption controller already prints out a message when it receives an add/update event from the informer. You can filter them in the logs by searching for "disruption.go". It shows that an add event was sent for the pod but never an update event. I also realized that the pod appearing in the cache (after the build update was received) doesn't mean that an update event has happened. First an add event needs to happen. 

In short, I think that the right thing to do is to increase the timeout. Let me know if you think it should be a different number.

**Comment by bparees** [comment]
the way this test is structured, this change means we're going to wait 60s every time it runs.  it'd be good if we can find a different solution.

what it should be doing is:

wait up to N(60) seconds for build to be in expected state
wait always M(10) additional seconds to ensure no additional state changes occur

what it is doing with your change:
wait always N(60) seconds, confirm build is in expected state

**Comment by csrwng** [comment]
Yup, you're right. We're also using a single timeout for all state changes, shouldn't we reset after each state change?

**Comment by bparees** [comment]
it is reset each time through the test.States loop, no?

**Comment by csrwng** [comment]
Sorry, was looking at it wrong. Yup, so just need to change the way that it waits for the state change.

**Comment by csrwng** [comment]
[test]

**Comment by csrwng** [comment]
@bparees I've updated the test to timeout in the way we described it. However, latest run is failing because a build is never marked as succeeded. I don't know how we can expect it to:
```
I0426 16:49:07.339079   22514 panics.go:76] PUT /api/v1/namespaces/integration/pods/mock-build8chf1-build/status: (2.770556ms) 200 [[integration.test/v1.5.2+43a9be4 (linux/amd64) kubernetes/43a9be4] 127.0.0.1:35260]
I0426 16:49:07.339539   22514 controller.go:117] Handling update of build pod integration/mock-build8chf1-build
I0426 16:49:07.339468   22514 replication_controller.go:378] Pod mock-build8chf1-build updated, objectMeta {Name:mock-build8chf1-build GenerateName: Namespace:integration SelfLink:/api/v1/namespaces/integration/pods/mock-build8chf1-build UID:c492877e-2ac1-11e7-a528-0e032702e8d8 ResourceVersion:487 Generation:0 CreationTimestamp:2017-04-26 16:48:57 -0400 EDT DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[openshift.io/build.name:mock-build8chf1] Annotations:map[openshift.io/scc:privileged openshift.io/build.name:mock-build8chf1] OwnerReferences:[] Finalizers:[] ClusterName:} -> {Name:mock-build8chf1-build GenerateName: Namespace:integration SelfLink:/api/v1/namespaces/integration/pods/mock-build8chf1-build UID:c492877e-2ac1-11e7-a528-0e032702e8d8 ResourceVersion:496 Generation:0 CreationTimestamp:2017-04-26 16:48:57 -0400 EDT DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[openshift.io/build.name:mock-build8chf1] Annotations:map[openshift.io/scc:privileged openshift.io/build.name:mock-build8chf1] OwnerReferences:[] Finalizers:[] ClusterName:}.
I0426 16:49:07.339577   22514 controller.go:162] Failing build integration/mock-build8chf1 because the pod has no containers
I0426 16:49:07.339587   22514 controller.go:202] Updating build integration/mock-build8chf1 status Running -> Failed (BuildPodDeleted)
```

Because the pod is not scheduled it doesn't have any container statuses, and we mark the build as failed. However, in the test, we expect the build's status to be complete:
https://github.com/openshift/origin/blob/master/test/common/build/controllers.go#L186-L187
How does it succeed today?

**Comment by bparees** [comment]
the test itself updates the build pod's status:
https://github.com/openshift/origin/blob/master/test/common/build/controllers.go#L263

(and then the build should be updated when the buildpodcontroller sees the build pod status change)

**Comment by csrwng** [comment]
Yes, but even if the pod's status is succeeded, if there's no container status(es) attached to the pod, then we mark the build as failed in the controller:
https://github.com/openshift/origin/blob/master/pkg/build/controller/buildpod/controller.go#L154-L159

It seems that along with updating the status to PodSucceeded we also need to set something in pod.Status.ContainerStatuses

**Comment by csrwng** [comment]
I see why this succeeded before. The logic was flawed... the `stateReached` variable was declared outside of the loop for each state we expected to reach. (https://github.com/openshift/origin/blob/master/test/common/build/controllers.go#L209) So it was enough to reach the first state and then we never cleared the variable. After the first success, this condition would never be true: https://github.com/openshift/origin/blob/master/test/common/build/controllers.go#L302

**Comment by bparees** [comment]
nice catch

**Comment by bparees** [comment]
squash and lgtm

**Comment by csrwng** [comment]
[merge]

**Comment by csrwng** [comment]
flake https://github.com/openshift/origin/issues/13183
[merge]

**Comment by csrwng** [comment]
[merge]

**Comment by bparees** [comment]
flake https://github.com/openshift/origin/issues/13966
[merge]

**Comment by csrwng** [comment]
flake https://github.com/openshift/origin/issues/11452
[merge]
